File: Week 11 (SuperZone)/Day 1/D_Friends_and_the_Restaurant.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long int
int main()
{
    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;
        vector<ll> v(n, 0);
        for (ll i = 0; i < n; i++)
        {
            ll x;
            cin >> x;
            v[i] -= x;
        }
        for (ll i = 0; i < n; i++)
        {
            ll x;
            cin >> x;
            v[i] += x;
        }
        sort(v.begin(), v.end(), greater<ll>());
        ll l = 0, r = n - 1;
        ll ans = 0;
        while (l < r)
        {
            if (v[l] > -1 && v[r] > -1)
            {
                ans++;
                l++;
                r--;
            }
            else if (v[l] < 0 && v[r] < 0)
            {
                break;
            }
            else if (v[l] > -1)
            {
                if (v[l] >= abs(v[r]))
                {
                    ans++;
                    r--;
                    l++;
                }
                else
                {
                    r--;
                }
            }
        }
        cout << ans << endl;
    }
}

File: Week 11 (SuperZone)/Day 1/D_Friends_and_the_Restaurant.java
import java.util.Comparator;
import java.util.Scanner;
import java.util.Vector;
import java.util.Collections;

public class D_Friends_and_the_Restaurant {
    public static void main(String[] args) {
        Scanner in=new Scanner(System.in);
        long t;
        t=in.nextLong();
        for(long i=0;i<t;i++)
        {
            long defaultValue = 0L;
            int n;
            n=in.nextInt();
            Vector<Long> v = new Vector<>(Collections.nCopies(n, defaultValue));
            for(int j=0;j<n;j++)
            {
                long x=in.nextLong();
                v.set(j,v.get(j)-x);
            }
            for(int j=0;j<n;j++)
            {
                long x=in.nextLong();
                v.set(j,v.get(j)+x);
            }
            Collections.sort(v, new Comparator<Long>() {

                public int compare(Long o1, Long o2) {
                    return o2.compareTo(o1); // Descending order
                }
            });
            int l = 0, r = n - 1;
            long ans = 0;
            while (l < r)
            {
                if (v.get(l) > -1 && v.get(r) > -1)
                {
                    ans++;
                    l++;
                    r--;
                }
                else if (v.get(l) < 0 && v.get(r) < 0)
                {
                    break;
                }
                else if (v.get(l) > -1)
                {
                    if (v.get(l) >= Math.abs(v.get(r)))
                    {
                        ans++;
                        r--;
                        l++;
                    }
                    else
                    {
                        r--;
                    }
                }
            }
           System.out.println(ans);

        }
    }
}
